### PR TITLE
Updating python_requires and classifiers

### PR DESCRIPTION
**Pull request by acabarbaye**
Updating python_requires and classifiers
Fixed a python 3 issue in circle_tests

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,41 +12,60 @@
...
-        'Programming Language :: Python :: 3.4',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
-        "License :: OSI Approved :: Apache Software License",
-        "Operating System :: OS Independent",
+        'Programming Language :: Python',
+        'Topic :: Internet',
+        'Topic :: Software Development :: Embedded Systems',
+        'Topic :: Software Development :: Object Brokering',
```
Comment (mark-edgeworth) [inline setup.py]: Is this relevant?

File: MANIFEST.in [hunk 2]
```diff
@@ -1,2 +1,5 @@
-include README.md LICENSE pypi_readme.rst
+global-exclude *.py[cod] __pycache__ *.so
 graft test
+include README.rst
+include LICENSE
+include requirements.txt
```
Comment (mark-edgeworth) [inline MANIFEST.in]: Missing newline at the end of this file

File: setup.py [hunk 3]
```diff
@@ -12,41 +12,60 @@
...
-        "pyserial>=3.0,<4.0",
-        "mbed-os-tools>=0.0.9,<0.1.0",
-        "mercurial>=5.2"
-    ]
+    description="Command line tool for interacting with Mbed OS projects",
+    keywords="Mbed OS CLI",
+    include_package_data=True,
+    install_requires=requirements,
+    license='Apache 2.0',
+    long_description=long_description,
```
Comment (madchutney) [inline setup.py]: If we add this we can combine the `readme.md` and `readme.rst`
```suggestion
    long_description=long_description,
    long_description_content_type='text/markdown',
```

File: README.rst [hunk 4]
```diff
@@ -1,4 +1,4 @@
-.. image:: https://circleci.com/gh/ARMmbed/mbed-cli.svg?style=svg
+## Arm Mbed CLI
```
Comment (madchutney) [inline README.rst]: This isn't correct restructured text formatting, it's markdown. However, PyPI now supports markdown so we can combine the readmes in a single markdown file which works in both GitHub and PyPI.

File: setup.py [hunk 5]
```diff
@@ -12,41 +12,60 @@
...
+__version__ = None
+
+repository_dir = os.path.dirname(__file__)
+
+# single source for project version information without side effects
+with open(os.path.join(repository_dir, 'mbed', '_version.py')) as fh:
+    exec(fh.read())
+
+# .rst readme needed for pypi
+with open(os.path.join(repository_dir, 'README.rst')) as fh:
```
Comment (madchutney) [inline setup.py]: Could use the markdown version

File: requirements.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+pyserial>=3.0,<4.0
```
Comment (madchutney) [inline requirements.txt]: 👍 This will makes the installed much simpler

**Pull request merged**